File: Cellua/Api/Common/TextObject.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.Immutable;
using SFML.Graphics;
using SFML.System;

namespace Cellua.Api.Common
{
    public class TextObject
    {
        public readonly Text Text;

        public TextObject(Text text)
        {
            Text = text;
        }

        public void UseFont(FontObject font)
        {
            Text.Font = font.Font;
        }

        public IEnumerable<float> GetPosition() => new []{Text.Position.X, Text.Position.Y};

        public void SetPosition(float x, float y)
        {
            Text.Position = new Vector2f(x, y);
        }

        public void SetDisplayString(string s)
        {
            Text.DisplayedString = s;
        }

        public string GetDisplayString() => Text.DisplayedString;

        public void SetFillColor(byte r, byte g, byte b, byte a)
        {
            Text.FillColor = new Color(r, g, b, a);
        }

        public Tuple<byte, byte, byte, byte> GetFillColor()
        {
            return new Tuple<byte, byte, byte, byte>(Text.FillColor.R,
                Text.FillColor.G,
                Text.FillColor.B,
                Text.FillColor.A);
        }

    }
}

File: Cellua/ApiSpec/TextObject.lua
---@class TextObject
TextObject = {}


--- Changes the text font
---@param font FontObject
function TextObject.UseFont(font) end


--- Returns the position of the text in screen
---@return table x & y positions
function TextObject.GetPosition() end


--- Changes the position of the text
---@param x number x position of the text
---@param y number y position of the text
function TextObject.SetPosition(x, y) end


--- Changes the display string of the text
---@param s string string to display
function TextObject.SetDisplayString(s) end

--- Returns the display string of the text
---@return string display string
function TextObject.GetDisplayString() end


--- Changes the text fill color
---@param r number red
---@param g number green
---@param b number blue
---@param a number alpha
function TextObject.SetFillColor(r,g,b,a) end


--- Returns the text fill color
---@return number,number,number,number rgba
function TextObject.GetFillColor() end